tests/integration/container_runtime.py

```diff
@@ -83,11 +85,138 @@ def __init__(self) -> None:
+        """Initialize buildah runtime."""
+        self.name = "buildah"
+
+    def _configure_buildah_container(self, image: str) -> str:
+        """Configure buildah container.
+
+        Buildah requires a container to be explicitly created from an image before it can be used.
+        The built container then can be referenced by the name given when creating it.
+
+        :param image: The image to create a container from.
+        :return: The container name.
+        """
+        container_name = f"hermeto-test-container-{secrets.token_hex(6)}"
+        from_cmd = ["buildah", "from", "--name", container_name, image]
+        _, exit_code = _run_cmd(from_cmd)
+
+        if exit_code != 0:
+            raise RuntimeError(f"Failed to create buildah container from image {image}.")
+
+        return container_name
+
+    def _generate_cmd(
+        self, container_name: str, image: str, cmd: list[str], flags: Optional[list[str]]
+    ) -> list[str]:
+        """Generate container run command.
+
+        Buildah does not automatically use the image's configured CMD or ENTRYPOINT when running a command.
+        This means that the run command needs to be generated manually considering the image's configuration
+        and the flags present in the original run command.
+
+        :param container_name: The container name to run the command on.
+        :param image: The image on which the command will be run.
+        :param cmd: The original command provided.
+        :param flags: The flags present in the run command.
+        :return: The generated command.
+        """
+        filtered_flags, entrypoint = self._filter_entrypoint_flags(flags)
+        image_cmd, image_entrypoint = self._get_image_config(image)
+
+        # fallback to the image's default cmd if a custom one is not provided
+        cmd = cmd or image_cmd
+
+        # if an entrypoint flag is provided, use it
+        if entrypoint:
+            return ["run", *filtered_flags, container_name, "--", entrypoint, *cmd]
+
+        # if the image has an entrypoint, prepend it to the command
+        if image_entrypoint:
+            return ["run", *filtered_flags, container_name, "--", *image_entrypoint, *cmd]
+
+        return ["run", *filtered_flags, container_name, "--", *cmd]
+
+    def _filter_entrypoint_flags(
+        self, flags: Optional[list[str]]
+    ) -> tuple[list[str], Optional[str]]:
+        """Remove and return any entrypoint flags.
+
+        Since buildah does not support the --entrypoint flag, we need to extract it so that
+        the run command can be properly formatted later on.
+
+        :param flags: The flags to treat.
+        :return: The treated flags and the entrypoint value.
+        """
+        if flags is None:
+            return [], None
+
+        # find the index of the entrypoint flag, if it exists
+        index = next((i for i, flag in enumerate(flags) if flag.startswith("--entrypoint")), None)
+
+        if index is not None:
```

Comment:
`if index:`?

Reply:
No, it's not ok. :rabbit: has rightfully pointed out that if `index = 0`, this condition would fail in case it was as simple `if index`